nuxeo-core/nuxeo-core-query/src/main/java/org/nuxeo/ecm/platform/query/api/PageProvider.java

```diff
@@ -450,4 +450,9 @@
      * @since 9.1
      */
     void setHighlights(List<String> highlights);
+
+    /**
+     * @since 9.3
+     */
+    long getMaxResults();
```

Comment:
please add proper javadoc because people may be confused about what this represents